chasm/lib/scheduler/scheduler_tasks.go

```diff
@@ -35,15 +35,21 @@ func (r *SchedulerIdleTaskExecutor) Execute(
 func (r *SchedulerIdleTaskExecutor) Validate(
 	ctx chasm.Context,
 	scheduler *Scheduler,
-	task chasm.TaskAttributes,
-	_ *schedulerpb.SchedulerIdleTask,
+	taskAttrs chasm.TaskAttributes,
+	task *schedulerpb.SchedulerIdleTask,
 ) (bool, error) {
 	idleTimeTotal := r.Config.Tweakables(scheduler.Namespace).IdleTime
-	idleExpiration, isIdle := scheduler.getIdleExpiration(ctx, idleTimeTotal, time.Time{})
 
+	// Validate total idle time. If the time has been increased, we'll use the longer
```

Comment:
But this doesn't reschedule a new timer... So you just have to accept the time that was recorded when the task was generated (what I would do because it is simple) or you have to deal with this and when you execute the task, you reschedule a new task with the updated duration.